create-custom-map-style/README.md

```diff
@@ -0,0 +1,904 @@
+        "test": "echo \"Error: no test specified\" && exit 1"
+      },
+      "keywords": [],
+      "author": "",
+      "license": "ISC"
+    }
+    ```
+
+1. Next we need to add the Amazon Location Service API as a data source. However, in order to do this we need to use a local proxy that can handle SigV4 (TL;DR a way to provide your AWS credentials with http requests). For this we will use [tessera](https://github.com/mojodna/tessera) and [tilelive-aws](https://github.com/beatleboy501/tilelive-aws). From a new terminal (the `maputnik` process will be occupying the other terminal) run the following:
+
+    ```bash
+
+    nvm use 12
+    npm i tessera tilelive-aws aws4
+    ```
+
+1. The module makes use of several environment variables to be able to sign calls to the AWS API. You can run the following to generate the required AWS credentials:
+
+    ```bash
+
+    $ aws sts get-session-token --duration-seconds 36000 --profile Amplify-Default
+
+    {
+      "Credentials": {
+        "AccessKeyId": "XXXX0XX0X0XXXXXX0XXX",
+        "SecretAccessKey": "x0XXxxXXXxXxXX0xXxxX0XXxX0X0xxxx0XxxxxXX",
+        "SessionToken": "XXxXx0XXxX0xxX0XxXXx//////////....=",
+        "Expiration": "2021-00-00T00:00:00+00:00"
+      }
+    }
+    ```
+
+    Set these environment variables in your terminal like so:
+
+    ```bash
+
+    export AWS_ACCESS_KEY_ID=<Access Key from Previous Step>
+    export AWS_SECRET_ACCESS_KEY=<Secret Key from Previous Step>
+    export AWS_SESSION_TOKEN=<Session Token from Previous Step>
+    ```
+
+    or optionally run the helper script we've included (be sure to have jq installed first, `brew install jq` on Mac, or follow instructions for other platforms here: <https://stedolan.github.io/jq/download/>):
+    `./setAwsEnvVariables.sh Amplify-Default`
+
+1. This should be enough to get started using `tessera`. You'll need to provide the full file path to your module, so from your terminal run:
+
+    ```bash
+
+    npx tessera -r $(pwd)/node_modules/tilelive-aws/tilelive-aws.js aws:///CreateCustomMapStyle-<YOUR ENVIRONMENT NAME>
+    ```
+
+1. You should see some output in sdtout like `Listening at http://0.0.0.0:8080`.
+
+1. Now go back to Maputnik in your browser and click "Data Sources" in the top navigation bar. There should not be any active data sources. If you see any be sure to delete them.
+
+ ![Data Sources](media/MaputnikDataSources.png)
+ *Data Sources*
+
+1. We're going to add a new data source as shown below.
+
+- `Source ID`: `amazon`
+- `Source Type`: `Vector (XYZ URLs)`
+- `1st Tile URL`: `http://localhost:8080/{z}/{x}/{y}@2x.png`
+- `Min Zoom`: `0`
+- `Max Zoom`: `22`
+
+ ![New Data Source](media/MaputnikNewSource.png)
+ *New Data Source*
+
+1. In addition to Tiles, the type of web map we will be styling also makes use of Glyphs and Sprites. We can't use Tessera to proxy the endpoints for these, but we can however generate a signed URL for each endpoint to stick in our style descriptor JSON file.
```

Comment:
Does the signed URL work?

Reply:
Oops I did forget to update this, good catch. The signed URL doesn't work (or at least I wasn't able to get it to work)

Reply:
That doesn't surprise me, since I don't think AWS includes support for signing URL templates (i.e. things that will vary somehow).